manager/orchestrator/taskreaper/task_reaper.go

```diff
@@ -95,7 +96,7 @@ func (tr *TaskReaper) Run(ctx context.Context) {
 				}] = struct{}{}
 			case api.EventUpdateTask:
 				t := v.Task
-				if t.Status.State >= api.TaskStateOrphaned && t.ServiceID == "" {
+				if t.Status.State >= api.TaskStateOrphaned {
```

Comment:
This will only fire when the task actually gets updated, right? If we load this task out of a snapshot, this event will not fire for the task.

Reply:
Good point. I'll address that and see if I can add a test for that case.